api/v1beta1/azuremanagedcontrolplane_webhook_test.go

```diff
@@ -1583,6 +1584,29 @@ func TestAzureManagedControlPlane_ValidateUpdate(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "NetworkPolicy is allowed when NetworkPluginMode is not changed",
```

Comment:
There's a permutation we're not covering in UT:

1. Old = some NetworkPolicy + overlay NetworkPlugin
2. New = some different NetworkPolicy + overlay NetworkPlugin

I assume that AKS would reject that update, do we have the right webhooks to optimize for CAPZ UX?

Reply:
We already have webhook and test coverage for the immutability of NetworkPlugin which isn't affected by the value of NetworkPluginMode. Or do you think it's worth checking to see if some updates might actually be allowed when overlay is enabled?

Reply:
s/NetworkPlugin/NetworkPolicy/

Reply:
The subtlety of this change just makes me want to cover every (potentially) artificial scenario.

If we're fixing a specific use case let's just ship this and then we can watch closely and fine-tune our webhooks as we observe edge cases. We're already in a place where the actual behaviors don't match AKS documentation so I think we're probably just gonna have to be reactive. :/